File: backend/app/models/reminder.py
from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Reminder(Base):
    __tablename__ = "reminders"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), nullable=False, index=True)
    description = Column(String(1000), nullable=True)
    tags = Column(String(500), nullable=True)
    due_date = Column(DateTime(timezone=True), nullable=True)
    completed = Column(Boolean, default=False, nullable=False)
    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False
    )
    is_deleted = Column(Boolean, default=False, nullable=False)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)

    owner = relationship("User", back_populates="reminders")

File: frontend/src/types/reminder.ts
export type Reminder = {
  id: number
  title: string
  description: string | null
  tags: string | null
  due_date: string | null
  completed: boolean
  created_at: string
  updated_at: string
}

export type ReminderPayload = {
  title: string
  description: string | null
  tags: string | null
  due_date: string | null
  completed: boolean
}

export type ReminderQueryParams = {
  skip?: number
  limit?: number
  search?: string
  sort_by?: 'created_at' | 'updated_at' | 'due_date' | 'title'
  sort_dir?: 'asc' | 'desc'
}
